va/va.h

```diff
@@ -4885,6 +4895,47 @@ typedef enum
 #define VA_RENDER_DEVICE_LOCAL      1
 #define VA_RENDER_DEVICE_EXTERNAL   2
 
+/**\brief memory region info
+ * On chip memory may be devided to several different regions.
+ * vaDisplay should be attached with one memory region. All resource
+ * is allocated in this memory region. If the vaContext of one memory
+ * region access another resource in another memory region, it maybe
+ * will impact performance potentially.
+ * If the value == VA_ATTRIB_NOT_SUPPORTED, it mean that the attribute
+ * is unsupport or UNKNOWN.
+ * If application does not query this attribute or set this attribute
+ * the driver should make it transparent to application. Dirver could
+ * decide the resource location internally.
+ */
+
+typedef union _VADisplayAttribValMemoryRegion{
+    struct{
+        /** \brief current region index read - write */
+        uint32_t current_memory_region     : 4;
+        /** \brief memory region count, read - only */
+        uint32_t local_memory_regions      : 4;
+        /** \brief reserved bits for future, must be zero*/
+        uint32_t reserved  :8;
+        /** \brief bit mask to indicate which memory_region is available, read only
+         * \code
+         * VADisplayAttribValMemoryRegion reg;
+         * VADisplayAttribute reg_attr;
+         * reg_attr.type = VADisplayAttribMemoryRegion;
+         * vaGetDisplayAttributes(dpy, &reg_attr, 1);
+         * reg.value = reg_attr.value;
+         *
+         * for(int i = 0; i < reg.bits.local_memory_regions; i ++ ){
+         *    if((1<<i) & reg.bits.memory_region_masks){
+         *        printf("memory region %d can be selected", i);
+         *    }
+         *}
+         * \endcode
+         */
+        uint32_t memroy_region_masks : 16;
```

Comment:
fix typos: s/memroy/memory/. And should we use plural in a different way: `memory_regions_mask`?